### ISSUE TITLE
Should published lessons be moved from 'submissions' site

### ISSUE THREAD

**Issue by acrymble** [programminghistorian/jekyll#398]
Issue was raised over email to editorial board. Will the github submissions repository get too cluttered if we don't move completed lessons? 

How do we ensure moved lessons still work (internal links, link to github review tickets).

How can we ensure any extra work doesn't outweigh any extra benefit to moving?

**Comment by wcaleb** [comment]
I suggest we put them in a "published" folder and edit the YAML to include a `redirect_from` the original URL (without `published` in the path) so that the live previews mentioned in review tickets still work.

**Comment by mdlincoln** [comment]
I second moving them to a dedicated folder. This would also let them be excluded from html-proofer checks (which should eventually be added to ph-submissions like they have been to the live site).

**Comment by ianmilligan1** [comment]
Thirded (?).

**Comment by wcaleb** [comment]
I've run into a bit of a problem here. If I move a published lesson into a new folder on the `ph-submissions` repo, then we lose the whole commit history for the file. (Unfortunately, this means we've already lost the commit history for [the lesson that I have test moved](https://github.com/programminghistorian/ph-submissions/tree/gh-pages/lessons/published).

This raises two questions, probably for @mdlincoln:

1. Is it somehow possible for me to restore the repo such that the commit history for that file can be restored? (Quick glance at Stackoverflow suggests [the answer may be no](https://stackoverflow.com/questions/2314652/is-it-possible-to-move-rename-files-in-git-and-maintain-their-history).)
2. Is it important for us to keep the commit histories? My recollection is that we thought it was, as a way of documenting the whole revision and review process. Otherwise, there's no real reason to keep the published lessons in the ph-submissions repo at all, unless people want to come back and study the changes that were made in response to reviewer suggestions.

TL;DR - If commit history is important to us, then our plan to move published lessons into their own subfolder may have a fatal flaw.

**Comment by mdlincoln** [comment]
You're right that git has rather lossy behavior for tracking moved/renamed files. The file history will be less visible on GitHub, because it uses the basic `git log` function. Someone running `git log --follow` on the other hand, can still access the full change history. I did that just now, and can see the full change history even after you moved the file. So, it's still accessible, but you need to use the command line. **That is definitely a high barrier!** But then again, for the (low?) numbers of people going through old revisions, that may be an acceptable barrier.

My suggestion is not to worry about it too much. _PH_ already made a decisive tradeoff in favor of usability/access, and against robust change tracking, by shifting away from pull requests in favor of a separate ph-submissions repo. For the diversity of our user base (which includes authors, reviewers, and our own editors) I think that is still the right decision.

**Comment by wcaleb** [comment]
@mdlincoln, out of idle curiosity, I looked to see if links to old commits worked even if the file is moved. They do! See, for example:

https://github.com/programminghistorian/ph-submissions/blob/d4de42d8876f718eab48897bc40cb84499cb4c8e/lessons/using-javascript-to-create-maps.md

From there you can click on "history" and actually see the commit history of the old file. So this means (a) I owe you even MORE of an apology for taking up time on this during the call! and (b) we can still have our cake and eat it, too, at least in cases where a commit on a lesson referenced the ticket number.

At any rate, I'll move forward on getting this folder implemented so we can close this ticket.

**Comment by mdlincoln** [comment]
Oh huh, I think that might actually be a relatively new GH feature - even a month ago I felt like that didn't work on a different jekyll site I work on. Anyway, great.

See https://github.com/programminghistorian/ph-submissions/issues/80 - I also noted on there that images should be moved as well.
